src/Classes/TradeQueryRequests.lua

```diff
@@ -143,21 +143,27 @@ function TradeQueryRequestsClass:SearchWithQueryWeightAdjusted(realm, league, qu
 				self:FetchResults(response.result, response.id,  callback)
 			end
 		else
-			-- Search clipped, fetch highest weight item, update query weight and repeat search
-			previousSearchItemIds = response.result
-			previousSearchId = response.id
-			local firstResultBatch = {unpack(response.result, 1, math.min(#response.result, 10))}
-			self:FetchResults(firstResultBatch, response.id, function(items, errMsg)
-				if errMsg then
-					return callback(nil, errMsg)
-				end
-				previousSearchItems = items
-				local highestWeight = items[1].weight
+			if response.total < self.maxFetchPerSearch then -- Less than maximum items retrieved lower weight to try and get more.
 				local queryJson = dkjson.decode(query)
-				queryJson.query.stats[1].value.min = (tonumber(highestWeight) + queryJson.query.stats[1].value.min) / 2
+				queryJson.query.stats[1].value.min = queryJson.query.stats[1].value.min / 10
```

Comment:
-90% decrease seems pretty agressive compared to +50% adjustment below.
This may cause overshooting and slow convergence.

Reply:
If it overshoots the average thing seems very effective of getting it in the desired range so it would only overshoot once.
But it does seem to work fine with division by 2 if that is preferable.
If you let it overshoot you get thousands of results which could be desirable vs maybe a few hundred which is nice for people want to refine the search themselves. I am not opposed to making this less aggressive if people think that is more reasonable.